### PR TITLE
sdk changes for SuperMinterV2

### PR DESCRIPTION
**Pull request by vigneshka**

### CODE REVIEW

File: packages/sdk/src/utils/cache-singleton.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+export class CacheSingleton {
+  private static instance: CacheSingleton
+  private cache: Map<string, any>
```
Comment (PabloSzx) [inline packages/sdk/src/utils/cache-singleton.ts:3]: nah, don't use `any`

**Review by PabloSzx — CHANGES_REQUESTED** [review]

File: packages/sdk/src/utils/cache-singleton.ts (hunk 1 above)
Comment (PabloSzx) [reply packages/sdk/src/utils/cache-singleton.ts:3]: you can use `unknown`

Comment (PabloSzx) [reply packages/sdk/src/utils/cache-singleton.ts:3]: or to have guaranteed on type-safety just make the cache singleton to hold the type, which has to be the same on setter and getter

Comment (vigneshka) [reply packages/sdk/src/utils/cache-singleton.ts:3]: unknown causes issues with this:

export class CacheUtils {
  private static cache = CacheSingleton.getInstance()

  public static async getOrSetCache<T>(key: string, fetchData: () => Promise<T>): Promise<T> {
    let data = this.cache.get(key)
    if (!data) {
      data = await fetchData()
      this.cache.set(key, data)
    }
    return data
  }
}

how can I make this general purpose without any?

File: packages/sdk/src/utils/cache-utils.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export class CacheUtils {
+  private static cache = new Map<string, unknown>()
+
+  public static async getOrSetCache<T>(key: string, fetchData: () => Promise<T>): Promise<T> {
+    if (this.cache.has(key)) {
+      return this.cache.get(key) as T
+    }
+    const data = await fetchData()
```
Comment (PabloSzx) [inline packages/sdk/src/utils/cache-utils.ts:8]: I still feel like making this call idempotent would be worth

File: packages/sdk/src/contract/edition-v2/read/mint.ts [hunk 3]
```diff
@@ -41,43 +39,39 @@ export type GetTotalMintPriceAndFeesReturnType = {
...
+      const address = superMinter.address
+      return client.readContract({
+        abi,
+        address,
+        functionName: 'totalPriceAndFees',
+        args: [editionAddress, tier, scheduleNum, quantity, false],
+      })
+    }
+    default:
+      exhaustiveGuard(version)
```
Comment (PabloSzx) [inline packages/sdk/src/contract/edition-v2/read/mint.ts:73]:
```suggestion
      exhaustiveGuard(superMinter)
```

File: .changeset/six-seas-approve.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@soundxyz/sdk': minor
+'@soundxyz/legacy-sdk': patch
+---
+
+support SuperMinterV1_1
```
Comment (vigneshka) [inline .changeset/six-seas-approve.md:6]:
```suggestion
support SuperMinterV2
```

**Review by PabloSzx — APPROVED** [review]

File: .changeset/six-seas-approve.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+'@soundxyz/sdk': minor
+'@soundxyz/legacy-sdk': patch
```
Comment (PabloSzx) [inline .changeset/six-seas-approve.md:3]:
```suggestion
```

**Pull request merged**